### ISSUE TITLE
Flush logs button in settings page has no effect

### ISSUE THREAD

**Issue by sparklyballs** [pi-hole/pi-hole#1823]
**In raising this issue, I confirm the following:** `{please fill the checkboxes, e.g: [X]}`

- [x] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [x] The issue I am reporting can be *replicated*.
- [x] The issue I am reporting isn't a duplicate (see [FAQs](https://github.com/pi-hole/pi-hole/wiki/FAQs), [closed issues](https://github.com/pi-hole/pi-hole/issues?utf8=%E2%9C%93&q=is%3Aissue%20is%3Aclosed%20), and [open issues](https://github.com/pi-hole/pi-hole/issues)).

**How familiar are you with the the source code relevant to this issue?:**

`1`

---
**Expected behaviour:**

Clicking the flush logs button 
![screen shot 2017-12-09 at 10 53 50](https://user-images.githubusercontent.com/4343808/33795010-4ab58486-dccf-11e7-896a-f39a00d40a85.png)
and clicking yes to the warning does not clear the logs/stats etc on the admin page

**Actual behaviour:**

The logs and stats on the admin page are not cleared using opera or safari and clearing caches forcing reloads etc

**Steps to reproduce:**

click the flush logs button and accept the warning by pressing yes

**Debug token provided by [uploading `pihole -d` log](https://discourse.pi-hole.net/t/the-pihole-command-with-examples/738#debug):**

`n81y7unxne`

**Troubleshooting undertaken, and/or other relevant information:**

`{Steps of what you have done to fix this}`

> * `{Please delete this quoted section when opening your issue}`
> * You must follow the template instructions. Failure to do so will result in your issue being closed.
> * Please [submit any feature requests here](https://discourse.pi-hole.net/c/feature-requests), so it is votable and trackable by the community.
> * Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.
> * Detail helps us understand and resolve an issue quicker, but please ensure it's relevant.
> * _This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

**Comment by sparklyballs** [comment]
![screen shot 2017-12-09 at 10 53 50](https://user-images.githubusercontent.com/4343808/33795032-e6410876-dccf-11e7-8232-95d7d70ec000.png)

**Comment by DL6ER** [comment]
Please run
```
killall -s SIGHUP pihole-FTL
```
and see if that clears the statistics on the web.

**Comment by sparklyballs** [comment]
i ran 

sudo killall -s SIGHUP pihole-FTL

and refreshed the web page and the stats still aren't cleared

**Comment by DL6ER** [comment]
Okay, I'm still trying to figure out what is causing the issue you are seeing. After you clicked on `Flush logs`, how does `/var/log/pihole.log` look like? Has it been emptied?

**Comment by sparklyballs** [comment]
no there are many lines in that file

is it safe (ie no identifiers to me etc)  to post it here as an attachment ?

**Comment by sparklyballs** [comment]
if it helps

`pi@pihole:~ $ stat /var/log/pihole.log
  File: /var/log/pihole.log
  Size: 5837903   	Blocks: 11416      IO Block: 4096   regular file
Device: b302h/45826d	Inode: 20740       Links: 1
Access: (0644/-rw-r--r--)  Uid: (  109/ dnsmasq)   Gid: (    0/    root)
Access: 2017-09-21 02:59:45.524138509 +0100
Modify: 2017-12-09 12:23:02.030734339 +0000
Change: 2017-12-09 12:23:02.030734339 +0000
 Birth: -`

**Comment by DL6ER** [comment]
```
Size: 5837903
```
sounds like 5 MB, okay, so flushing doesn't seem to work at all for you...

Does it work if you run
```
pihole -f
```
manually?

**Comment by sparklyballs** [comment]
that does work

``pi@pihole:~ $ pihole -f
  [✓] Flushed /var/log/pihole.log
pi@pihole:~ $ stat /var/log/pihole.log
  File: /var/log/pihole.log
  Size: 186       	Blocks: 8          IO Block: 4096   regular file
Device: b302h/45826d	Inode: 20740       Links: 1
Access: (0644/-rw-r--r--)  Uid: (  109/ dnsmasq)   Gid: (    0/    root)
Access: 2017-09-21 02:59:45.524138509 +0100
Modify: 2017-12-09 14:24:06.385248671 +0000
Change: 2017-12-09 14:24:06.385248671 +0000
 Birth: -``

**Comment by DL6ER** [comment]
Hmm, can you perform any other change from the Settings page?

**Comment by sparklyballs** [comment]
i can update the lists via the webui

![screen shot 2017-12-09 at 14 27 30](https://user-images.githubusercontent.com/4343808/33796488-3090abf8-dced-11e7-8a96-d043fe68829c.png)

**Comment by sparklyballs** [comment]
and rebooting  the system from the webui works too

**Comment by norrkin** [comment]
Can confirm that flush button via the GUI doesn't.

**Comment by ajp2k13** [comment]
Same problem here on my Pi. If I disable query logging first then flush, it works...

**Comment by DL6ER** [comment]
After a bit of tinkering I've been able to reach a configuration where I'm affected by this bug as well. Investigating...

**Comment by dschaper** [comment]
Closed via #1856

**Comment by trhsrw** [comment]
I have exactly the same issue in v 3.3 with exactly the same observations sparkyballs reported. I don't know how to reopen a ticket. So if this is not being modified within a week or so I will open a new ticket. I am using a raspi 2 B with the latest and up-to-date Raspbian.

**Comment by PromoFaux** [comment]
Slightly different with 3.3, but that is fixed in the `development` branch, and will go into the next release. No need to open a new ticket

**Comment by trhsrw** [comment]
ok, thanks for the quick answer and fix

**Comment by RIKWEBB** [comment]
try this :)
cd /etc/pihole
sudo service pihole-FTL stop
sudo rm pihole-FTL.db
sudo service pihole-FTL start

**Comment by DL6ER** [comment]
@dmblack You just happened to post to the wrong issue report as this time there was none on GitHub, only on [Discourse](discourse.pi-hole.net). The issue reported in here was indeed resolved with v3.2.1 but was entirely different from the issue resolved in v3.3.1 which was mentioned in https://github.com/pi-hole/pi-hole/pull/1929

**Comment by dmblack** [comment]
@DL6ER 

Not sure I understand what you are saying? #1929 does not clearly indicate the UI is impacted, though perhaps that's the command the UI is calling.

I do now see that there's a tag for milestone 3.2.1, though it's probably wise to be clear on a regression of a UI error.

Thanks,

**Comment by DL6ER** [comment]
> #1929 does not clearly indicate the UI is impacted, though perhaps the UI is only calling this function you reference anyway.

Yes

> this issue has been fixed (again?) in 3.3.1

No, they have been two *entirely* separate issues, which is what I tried to bring across earlier. The first issue (some time ago, this issue report) was about a defect on the web UI, not  being able to perform the action in the backend. The `pihole -f` command, however, was not affected then. The second issue (that has been fixed in the referenced PR #1929 ) was an altogether new issue in a different part of the code and this time affected both the CLI and the web UI.

I hope things are more clear now.
